flang/lib/Lower/DirectivesCommon.h

```diff
@@ -819,6 +819,13 @@ genBoundsOps(fir::FirOpBuilder &builder, mlir::Location loc,
         extent = fir::factory::readExtent(builder, loc, dataExv, dimension);
         if (mlir::isa<fir::UndefOp>(extent.getDefiningOp())) {
           extent = zero;
+          if (ubound && lbound &&
+              mlir::isa<mlir::arith::ConstantOp>(ubound.getDefiningOp()) &&
+              mlir::isa<mlir::arith::ConstantOp>(lbound.getDefiningOp())) {
```

Comment:
Thanks for the fix. You can remove the check for constant. If lb and ub are set we can compute the extent even if those are variables. `a(n:m)` should also have a proper extent compute and not zero. The +1 is correct.

Reply:
Sounds good, I'll remove the constant check!